### PR TITLE
webgpu: save built-in varialbes into global variables

### PR DESCRIPTION
**Pull request by xhcao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5945)

### CODE REVIEW

**Comment by xhcao** [comment]
@qjia7 @axinging @haoyunfeix @gyagp Please take a look, thank you.

File: tfjs-backend-webgpu/src/shader_preprocessor.ts [hunk 1]
```diff
@@ -597,10 +596,7 @@ export function generateGetOutputCoords(
   const outRank = outShape.length;
   if (x.length === outRank) {
     const dtype = getCoordsDataType(outRank);
-    const snippet =
-        `fn getOutputCoordsWithFlatDispatchLayout(globalId : vec3<u32>, localId : vec3<u32>, numWorkgroups: vec3<u32>) -> ${
-            dtype}{
-      let globalIndex = getGlobalIndex(globalId, localId, numWorkgroups);
+    const snippet = `fn getOutputCoords(globalIndex: i32) -> ${dtype}{
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/shader_preprocessor.ts:599]: I prefer `fn getOutputCoords()` instead of `fn getOutputCoords(globalIndex: i32)`

Comment (xhcao) [reply tfjs-backend-webgpu/src/shader_preprocessor.ts]: Done

**Review by qjia7 — APPROVED** [review]
LGTM, thanks.

**Comment by qjia7** [comment]
cc @lina128 @pyu10055

**Pull request merged**